lib/ain-ocean/src/data_acces/masternode.rs

```diff
@@ -1,19 +1,47 @@
-use anyhow::{anyhow, Result};
-use serde::{Deserialize, Serialize};
-use serde_json;
-
 use crate::{
-    database::db_manager::{ColumnFamilyOperations, RocksDB},
+    database::db_manager::{ColumnFamilyOperations, RocksDB, SortOrder},
     model::masternode::Masternode,
 };
+use anyhow::{anyhow, Result};
+use rocksdb::IteratorMode;
+use serde_json;
 
 pub struct MasterNodeDB {
     pub db: RocksDB,
 }
 
 impl MasterNodeDB {
-    pub async fn query(&self, limit: i32, lt: i32) -> Result<Vec<Masternode>> {
-        todo!()
+    pub async fn query(
+        &self,
+        limit: i32,
+        lt: u32,
+        sort_order: SortOrder,
+    ) -> Result<Vec<Masternode>> {
+        let iter_mode: IteratorMode = sort_order.into();
+        let master_node: Result<Vec<_>> = self
+            .db
+            .iterator("masternode", iter_mode)?
+            .into_iter()
+            .take(limit as usize)
+            .map(|result| {
+                result
+                    .map_err(|e| {
+                        anyhow!("Error during iteration: {}", e)
+                            .context("error master_node query error")
+                    })
+                    .and_then(|(_key, value)| {
+                        let stats: Masternode = serde_json::from_slice(&value)?;
+                        if stats.block.height < lt {
```

Comment:
I'm not sure this `lt` will do what we expect on both SortOrder.
I think we'd better add a `From<(SortOrder, StartKey)>` to `Itermode::From(<StartKey.as_slice()>, <SortOrder into Direction>)`, drop the `lt` altogether and replace it with the start index.

Reply:
I am facing issue whenever I use `lt` for Ascending order. Its doesn`t loop whole collection, its just throw the error. I ll update with `startKey`. Rocks db doesn't store data in order.

Reply:
Keys are sorted. That's why we need a secondary sort index by block height + txno to have an iteration that makes sense (Retrieve masternodes by order of creation).